packages/salesforcedx-vscode-core/src/commands/util/parameterGatherers.ts

```diff
@@ -161,6 +165,88 @@ export class DemoModePromptGatherer implements ParametersGatherer<{}> {
   }
 }
 
+export class SelectLwcComponentDir
+  implements ParametersGatherer<{ fileName: string; outputdir: string }> {
+  private typeDir: string;
+  private typeDirRequired: boolean | undefined;
+  public static readonly defaultOutput = path.join('main', 'default');
+
+  public constructor(typeDir: string, typeDirRequired?: boolean) {
+    this.typeDir = typeDir;
+    this.typeDirRequired = typeDirRequired;
+  }
+
+  public async gather(): Promise<
+    CancelResponse | ContinueResponse<{ fileName: string; outputdir: string }>
+  > {
+    let packageDirs: string[] = [];
+    try {
+      packageDirs = await SfdxPackageDirectories.getPackageDirectoryPaths();
+    } catch (e) {
+      if (
+        e.name !== 'NoPackageDirectoryPathsFound' &&
+        e.name !== 'NoPackageDirectoriesFound'
+      ) {
+        throw e;
+      }
+    }
+
+    const dirOptions = this.getDefaultOptions(packageDirs);
+    let outputdir = await this.showMenu(dirOptions);
+    const namePathMap = new Map();
+    let filepath;
```

Comment:
can we make this all const please? If there's a need to mutate, we can assign to a new var